### PR TITLE
Feature/2215 add api operations to work preferences form

### LINKED ISSUE #2215: Add API operations to Work Preferences form
# Description

Add api operations to populate initial values of form, and to submit values of form.

- Get values of Work Preferences from Applicant object to populate initial values.
- Update the User with new Work Preferences values when submitted. 
- Get/update the accepted OperationalRequirements from User model.
- WouldAcceptTemporary field.

[Wireframe](https://www.figma.com/proto/PrvXEDac0HVp7dF7SLWVFm/Candidate-profile-interfaces---V5.0.0?node-id=2485%3A5000&starting-point-node-id=1%3A2)

# Acceptance Criteria
- [ ] API integrated with Work Preferences form.
- [ ] API responds with success/failure toast message.
- [ ] Navigation to Applicant Profile on success.

### PR DESCRIPTION
**Pull request by RVany**
Resolves #2215 
# Description

Add api operations to populate initial values of form, and to submit values of form.

- Get values of Work Preferences from Applicant object to populate initial values.
- Update the User with new Work Preferences values when submitted. 
- Get/update the accepted OperationalRequirements from User model.
- WouldAcceptTemporary field.

[Wireframe](https://www.figma.com/proto/PrvXEDac0HVp7dF7SLWVFm/Candidate-profile-interfaces---V5.0.0?node-id=2485%3A5000&starting-point-node-id=1%3A2)

# Acceptance Criteria
- [x] API integrated with Work Preferences form.
- [x] API responds with success/failure toast message.
- [x] Navigation to Applicant Profile on success.

### CODE REVIEW

File: frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx [hunk 1]
```diff
@@ -1,26 +1,96 @@
...
+    return {
+      wouldAcceptTemporary: boolToString(data?.me?.wouldAcceptTemporary),
+      acceptedOperationalRequirements:
+        data?.me?.acceptedOperationalRequirements,
+    };
+  };
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => {
+    const StringtoBool = (stringVal: string): boolean | null | undefined => {
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx:69]: Super nitpicking here but can we change the case on this so it starts with a lower case "s" i.e `stringToBool`?

**Review by esizer — CHANGES_REQUESTED** [review]
Awesome work! It looks good outside of a *very* small change.

File: frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx (hunk 1 above)
Comment (RVany) [reply frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx:69]: Sure! Thank you!

**Review by esizer — APPROVED** [review]
Awesome! 😃

**Comment by RVany** [comment]
> Awesome! 😃

Thank you! I am not sure how that happend!

**Pull request merged**